homeassistant/components/tesla_fleet/climate.py

```diff
@@ -0,0 +1,300 @@
+        value = self.get("climate_state_is_climate_on")
+        if value is None:
+            self._attr_hvac_mode = None
+        elif value:
+            self._attr_hvac_mode = HVACMode.HEAT_COOL
+        else:
+            self._attr_hvac_mode = HVACMode.OFF
+
+        self._attr_current_temperature = self.get("climate_state_inside_temp")
+        self._attr_target_temperature = self.get(f"climate_state_{self.key}_setting")
+        self._attr_preset_mode = self.get("climate_state_climate_keeper_mode")
+        self._attr_min_temp = cast(
+            float, self.get("climate_state_min_avail_temp", DEFAULT_MIN_TEMP)
+        )
+        self._attr_max_temp = cast(
+            float, self.get("climate_state_max_avail_temp", DEFAULT_MAX_TEMP)
+        )
+
+    async def async_turn_on(self) -> None:
+        """Set the climate state to on."""
+
+        self.raise_for_signing()
+        self.raise_for_scope(Scope.VEHICLE_CMDS)
+        await self.wake_up_if_asleep()
+        await handle_vehicle_command(self.api.auto_conditioning_start())
+
+        self._attr_hvac_mode = HVACMode.HEAT_COOL
+        self.async_write_ha_state()
+
+    async def async_turn_off(self) -> None:
+        """Set the climate state to off."""
+
+        self.raise_for_signing()
+        self.raise_for_scope(Scope.VEHICLE_CMDS)
+        await self.wake_up_if_asleep()
+        await handle_vehicle_command(self.api.auto_conditioning_stop())
+
+        self._attr_hvac_mode = HVACMode.OFF
+        self._attr_preset_mode = self._attr_preset_modes[0]
+        self.async_write_ha_state()
+
+    async def async_set_temperature(self, **kwargs: Any) -> None:
+        """Set the climate temperature."""
+
+        if temp := kwargs.get(ATTR_TEMPERATURE):
+            await self.wake_up_if_asleep()
+            await handle_vehicle_command(
+                self.api.set_temps(
+                    driver_temp=temp,
+                    passenger_temp=temp,
+                )
+            )
+            self._attr_target_temperature = temp
+
+        if mode := kwargs.get(ATTR_HVAC_MODE):
+            # Set HVAC mode will call write_ha_state
+            await self.async_set_hvac_mode(mode)
+        else:
+            self.async_write_ha_state()
+
+    async def async_set_hvac_mode(self, hvac_mode: HVACMode) -> None:
+        """Set the climate mode and state."""
+        if hvac_mode == HVACMode.OFF:
+            await self.async_turn_off()
+        else:
+            await self.async_turn_on()
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set the climate preset mode."""
+        await self.wake_up_if_asleep()
+        await handle_vehicle_command(
+            self.api.set_climate_keeper_mode(
+                climate_keeper_mode=self._attr_preset_modes.index(preset_mode)
+            )
+        )
+        self._attr_preset_mode = preset_mode
+        if preset_mode == self._attr_preset_modes[0]:
+            self._attr_hvac_mode = HVACMode.OFF
```

Comment:
My impression is that turning off a preset mode should turn off a preset but not necessarily turn off the A/C. That is, it would just mean the A/C should be behaving normally without a special mode, but could still be on.

Is there a specific reason for this or is leaving the hvac mode alone here right?

Reply:
I think this might actually be an edge case bug. Turning off the present shouldn't turn off HVAC, but only if the user is still in the vehicle.

If they are not in the vehicle, the vehicle will turn off the HVAC when preset is off, which is why you typically use a preset like these anyway.

I'll rethink this.

Reply:
This line was removed, its a bug.

I think most users would never run into this because they wouldn't be using Home Assistant to modify the preset mode when sitting inside their vehicle, but its defintely possible and put the entity into the wrong state.